From 8c1708d1ff32a0303b59bb17fc3b95a7292eef7f Mon Sep 17 00:00:00 2001
From: "i.MX Yocto Project Release" <imx.release@nxp.com>
Date: Mon, 24 Oct 2022 16:21:22 -0700
Subject: [PATCH] core: psa: Fix build break

```
In function 'set_key_algo',
|     inlined from 'set_key_attributes_list' at core/psa/keymgr.c:506:11:
| core/psa/keymgr.c:447:17: error: 'kdf_tlv' may be used uninitialized [-Werror=maybe-uninitialized]
|   447 |                 smw_tlv_set_length(kdf_tlv, p);
|       |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| core/psa/keymgr.c: In function 'set_key_attributes_list':
| core/psa/keymgr.c:380:24: note: 'kdf_tlv' was declared here
|   380 |         unsigned char *kdf_tlv;
|       |                        ^~~~~~~
```

Signed-off-by: i.MX Yocto Project Release <imx.release@nxp.com>
---
 core/psa/keymgr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/psa/keymgr.c b/core/psa/keymgr.c
index 3cc5975..b295ad6 100644
--- a/core/psa/keymgr.c
+++ b/core/psa/keymgr.c
@@ -377,7 +377,7 @@ static int set_key_algo(psa_algorithm_t alg, unsigned char **tlv,
 	uint8_t length = 0;
 	uint8_t min_length = 0;
 	unsigned char *p;
-	unsigned char *kdf_tlv;
+	unsigned char *kdf_tlv = NULL;
 
 	SMW_DBG_TRACE_FUNCTION_CALL;
 
-- 
2.17.1

